cloudformation/extraction.yaml

```diff
@@ -48,9 +57,39 @@ Resources:
   ExtractQueueDeadLetterQueue:
     Type: AWS::SQS::Queue
     Properties:
-      QueueName: !Join [ '-', [!Ref Prefix, !Ref Environment, extract-dlq ] ]
+      QueueName: !Join ["-", [!Ref Prefix, !Ref Environment, extract-dlq]]
       KmsMasterKeyId: !GetAtt KmsKey.Arn
 
+  TextractRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              Service:
+                - "textract.amazonaws.com"
+            Action:
+              - "sts:AssumeRole"
+      Policies:
+        - PolicyName: "TextractRoleAccess"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "s3:GetObject*"
+                Resource: !GetAtt RawInvoicePdfBucket.Arn
+              - Effect: Allow
+                Action:
+                  - "sns:Publish"
+                Resource: !Ref AmazonTextractRawInvoiceDataTopic
+              - Effect: Allow
+                Action:
+                  - kms:*
```

Comment:
Better lock this down a little as this also opens up key-management like kms:DisableKey etc.

Reply:
Putting this in and then will deploy to check the textract-sns-sqs flow